addons/web_editor/models/ir_qweb.py

```diff
@@ -53,9 +53,11 @@ def _compile_directive_snippet(self, el, options):
         view_id = View.get_view_id(el.attrib.get('t-call'))
         name = View.browse(view_id).name
         thumbnail = el.attrib.pop('t-thumbnail', "oe-thumbnail")
-        div = u'<div name="%s" data-oe-type="snippet" data-oe-thumbnail="%s">' % (
+        forbid_sanitize = el.attrib.get('t-forbid-sanitize')
+        div = u'<div name="%s" data-oe-type="snippet" data-oe-thumbnail="%s"%s>' % (
```

Comment:
is this how it is usually done? I hate it :grimacing: 
Just add the space, we don't care if there is an useless trailing space if the `data-oe-forbid-sanitize` is not set

Reply:
Boh? How is that better than... just not have that space? :thinking:

Reply:
```py
"class='text-center%s'%s%s" % {
   ' text-red' if whatever else '',
   ' data-oe-forbid-sanitize="true"' if forbid_sanitize else '',
   " data-something='something'" if xx else "",
}
```
Is barely readable tbh. It is worth doing if we don't want to messes with the DOM for public user but here, no reason?
It is also bug prone, as everything relies on that ugly leading whitespace in every strings :grimacing: 
```py
"class='text-center %s' %s %s" % {
   'text-red' if whatever else '',
   'data-oe-forbid-sanitize="true"' if forbid_sanitize else '',
   "data-something='something'" if xx else "",
}
```

Reply:
That just doesn't feel like a detail for me TBH (maybe I am a weird person :grimacing: ), I feel like it changes everything visually and reading wise :o

Reply:
Went for the "easy to read" way - and will prefer it from now on.
In terms of personal taste, I'd rather never build XML from strings concatenation/substitution ever... but such as change is not going to happen in this project... :stuck_out_tongue: